py_mmd_tools/check_mmd.py

```diff
@@ -0,0 +1,213 @@
+"""
+Tools for checking if an XML MMD file is valid.
+
+License:
+     This file is part of the py-mmd-tools repository (https://github.com/metno/py-mmd-tools).
+     py-mmd-tools is licensed under GPL-3.0 (
+     https://github.com/metno/py-mmd-tools/blob/master/LICENSE)
+"""
+
+import logging
+import requests
+import pythesint as pti
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def check_rectangle(rectangle):
+    """
+    Check if element geographic extent/rectangle is valid:
+        - only 1 existing rectangle element
+        - rectangle has north / south / west / east subelements
+        - -180 <= min_lat <= max_lat <= 180
+        -    0 <= min_lon <= max_lon <= 360
+    Args:
+        rectangle: list of elements found when requesting node(s) geographic_extent/rectangle
+          (output of ET request findall)
+    Returns:
+        True / False
+    """
+
+    directions = dict.fromkeys(['north', 'south', 'west', 'east'], np.nan)
+
+    ok = True 
+    if len(rectangle) > 1:
+        logger.debug("NOK - Multiple rectangle elements in file.")
+        return False
+
+    for child in rectangle[0]:
+        # Remove namespace if any
+        if child.tag.startswith("{"):
+            child.tag = child.tag.split('}', 1)[1]
+        directions[child.tag] = float(child.text)
+
+    if not (-180 <= directions['west'] <= directions['east'] <= 180):
+        logger.debug('NOK - Longitudes not ok')
+        ok = False
+    if not (-90 <= directions['south'] <= directions['north'] <= 90):
+        logger.debug('NOK - Latitudes not ok')
+        ok = False
+    if not ok:
+        logger.debug(directions)
+
+    return ok
+
+
+def check_urls(url_list):
+    """
+    Check that a list of URLs is valid
+    Args:
+        url_list: list of URLs
+    Returns:
+        True / False
+    """
+
+    ok = True
+    for url in url_list:
+
+        try:
+            r = requests.get(url, timeout=10)
+            r.raise_for_status()
+            logger.debug(f'OK - {url}')
+            r.close()
+        except Exception as e:
+            logger.debug(f'NOK - {url}')
+            logger.debug(e)
+            ok = False
+
+    return ok
+
+
+def check_cf(cf_names):
+    """
+    Check that names are valid CF standard names
+    Args:
+        cf_names: list of names to test
+    Returns:
+        True / False
+    """
+    ok = True
+    for cf_name in cf_names:
+        try:
```

Comment:
this could be simplified by just letting the error go through. Then it would reduce to 3 lines:
```suggestion
        pti.get_cf_standard_name(cf_name)
    return True
```

Reply:
Yes it would be shorter. But the point of the function and the script is to test if a file or several files are valid. If we let the errors go, the script will stop at the first error. I think it's more useful to know at once all the errors that a file may contain. Am I making sense?

Reply:
aha, ok